setup/command/apply.go

```diff
@@ -16,18 +16,33 @@ func (f *CommandFactory) Apply() cli.Command {
 				Name:  "quick",
 				Usage: "skips verification checks that normally run after 'terraform apply' has completed",
 			},
+			cli.BoolTFlag{
+				Name:  "push",
+				Usage: "setting it to false skips pushing local tfstate to s3 (default: true)",
+			},
 		},
 		Action: func(c *cli.Context) error {
 			args, err := extractArgs(c.Args(), "NAME")
 			if err != nil {
 				return err
 			}
 
+			provider, err := f.newAWSProviderHelper(c)
+			if err != nil {
+				return err
+			}
+
 			instance := f.NewInstance(args["NAME"])
 			if err := instance.Apply(!c.Bool("quick")); err != nil {
 				return err
 			}
 
+			if c.Bool("push") {
+				if err := instance.Push(provider.S3); err != nil {
```

Comment:
Let's move the block:
```
provider, err := f.newAWSProviderHelper(c)
if err != nil {
    return err
}
```

into the `if c.Bool("push") {}` block. Reason being: we don't want the error from ` f.newAWSProviderHelper(c)` to occur if the user specifies `--push=false`